src/logging/BaseLogger.ts

```diff
@@ -127,20 +131,16 @@ export class BaseLogger {
         this._openItems.clear();
     }
 
-    _persistItem() {
-        throw new Error("not implemented");
-    }
+    abstract _persistItem(item: LogItem, filter?: LogFilter | null, forced?: boolean): void;
 
-    async export() {
-        throw new Error("not implemented");
-    }
+    abstract export(): void;
 
     // expose log level without needing 
     get level() {
```

Comment:
I think this should work here:
```suggestion
    get level(): typeof LogLevel {
```

Reply:
I can see that type inference already infers the type here. Should I still explicitly mention the return type?

Reply:
I see that I should based on other typescript files in the project 👍